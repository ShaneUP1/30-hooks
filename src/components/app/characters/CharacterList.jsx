import React from 'react';
import PropTypes from 'prop-types';
import SingleCharacter from './SingleCharacter';

const CharacterList = ({ characters }) => {
  const characterElements = characters.map((character) => (
    <li key={character.id}>
      <SingleCharacter
        name={character.name}
        species={character.species}
        image={character.image}
      />
    </li>
  ));

  return <ul data-testid="characters">{characterElements}</ul>;
};

CharacterList.propTypes = {
  characters: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      species: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired
    })
  ).isRequired
};

export default CharacterList;
