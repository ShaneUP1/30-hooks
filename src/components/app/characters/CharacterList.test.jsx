import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterList from './CharacterList';

describe('CharacterList component', () => {
  afterEach(() => cleanup());
  it('renders a list of characters', () => {
    const { asFragment } = render(<CharacterList
      characters={[{ name: 'shane', image: 'www.url.com', species: 'toad', id: 1 }]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
