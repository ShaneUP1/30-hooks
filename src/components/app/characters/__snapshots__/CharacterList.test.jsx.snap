// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList component renders a list of characters 1`] = `
<DocumentFragment>
  <ul
    data-testid="characters"
  >
    <li>
      <figure>
        <img
          alt="shane"
          src="www.url.com"
        />
        <figcaption>
          <h2>
            shane
          </h2>
          <p>
            toad
          </p>
        </figcaption>
      </figure>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`SingleCharacter component renders a single character 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="shane"
      src="www.url.com"
    />
    <figcaption>
      <h2>
        shane
      </h2>
      <p>
        toad
      </p>
    </figcaption>
  </figure>
</DocumentFragment>
`;
